### PR TITLE
Shortcuts for Android 7.1

### PR DESCRIPTION
**Pull request by Dafnik**
### First time contributor checklist

- [X] I have read [how to contribute](https://github.com/WhisperSystems/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [X] I have signed the [Contributor Licence Agreement](https://whispersystems.org/cla/)

### Contributor checklist
- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Oneplus 3, Android 7.1.1
- [X] My contribution is fully baked and ready to be merged as is
- [X] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the Fixes #1234 syntax
- [X] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in the commit message of my first commit

----------

### Description
This Pull Request adds two shortcuts, which are added into Android 7.1.x. There are two shortcuts. One for creating a group and another one for creating a conversation.

### CODE REVIEW

File: res/xml/shortcuts.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        <categories android:name="android.shortcut.conversation" />
+
+    </shortcut>
+
+    <shortcut
+        android:icon="@drawable/ic_forum_grey_32dp"
+        android:enabled="true"
+        android:shortcutId="static-shortcut-ChatCreate"
+        android:shortcutShortLabel="@string/conversation_list_fragment__fab_content_description"
+        android:shortcutLongLabel="@string/conversation_list_fragment__fab_content_description"
```
Comment (moxie0) [inline res/xml/shortcuts.xml]: Let's define new strings for these labels and the labels above

File: res/xml/shortcuts.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            android:targetPackage="org.thoughtcrime.securesms"
+            android:action="android.intent.action.VIEW"
+            android:targetClass="org.thoughtcrime.securesms.GroupCreateActivity" />
+
+        <categories android:name="android.shortcut.conversation" />
+
+    </shortcut>
+
+    <shortcut
+        android:icon="@drawable/ic_forum_grey_32dp"
```
Comment (moxie0) [inline res/xml/shortcuts.xml]: I'm not sure this is the best icon. I think the pencil FAB is probably more consistent.

Comment (Dafnik) [inline res/xml/shortcuts.xml]: For what Shortcut? Conversation or Group?

**Comment by nikwen** [comment]
My suggestion would be to use dynamic shortcuts as described in https://developer.android.com/guide/topics/ui/shortcuts.html to achieve something like the following:

**Comment by Dafnik** [comment]
This would be a possible privacy leak. The launcher would know the 3 latest contacts you write to. Or @moxie0 would do you think?

**Comment by moxie0** [comment]
I'm cool with dynamic shortcuts

**Comment by Dafnik** [comment]
So is this pull request ready for merging? @moxie0
I will add Dynamic Shortcuts later in a new Pull Request.

**Comment by moxie0** [comment]
I haven't reviewed it, let's go straight to dynamic shortcuts

**Comment by Dafnik** [comment]
Okay, i will have a look on it later.

**Comment by nikwen** [comment]
I would suggest adding a toggle to the privacy settings that let's users disable personalized shortcuts if they aren't comfortable sharing that information with their launcher. It would be similar to the anti-screenshot toggle and would result in them getting just the non-personal new conversation/new group toggles.

I know that the project's policy is to introduce as few settings options as possible but I do think that it makes sense here. Props to @Dafnik for thinking about that!

**Comment by Dafnik** [comment]
We could take the anti screenshot toggle entry also for this. So you haven't two option entries. Anti-Screenshot setting would also disable personalized shortcuts.

**Comment by nikwen** [comment]
@Dafnik I'm not a fan of mixing those together. Depending on the launcher I might disable the personalized shortcuts but I would always want to have screenshot functionality.

**Comment by Dafnik** [comment]
Okay

**Comment by Dafnik** [comment]
@moxie0, I can't find a list with all Conversations in Code. So, from where can I get the latest conversations? Thanks for your help.

**Comment by moxie0** [comment]
check out the direct share code

**Comment by Dafnik** [comment]
So, I added DynamicShortcuts. When you click on a shortcut it starts into the conversation. My problem is, profilpictures aren't shown and the name of the recipient is null. Do you know why @moxie0 ?

**Comment by Dafnik** [comment]
Finished DynamicShortcuts. Here is a little preview.
preview

**Comment by nikwen** [comment]
There seems to be a merge conflict in `src/org/thoughtcrime/securesms/database/ThreadDatabase.java`.

**Comment by Dafnik** [comment]
@moxie0, could you please answer the question, if we should make a custom privacy toggle for contact based shortcuts.

**Comment by moxie0** [comment]
@Dafnik No I would not like to include a new preference. I'm open to merging the behavior in with an existing preference if people feel strongly about that

**Comment by Dafnik** [comment]
I need to rebase the pull request and start from scratch.

**Comment by Dafnik** [comment]
@moxie0. When I create the Shortcuts in ConversationListActivity at onCreate() the Recipients aren't encrypted and the name and the picture is null. So in which method or class are the Recipients encrypted or loaded?

**Pull request closed without merging**